crates/swc/src/config/mod.rs

```diff
@@ -1057,52 +1057,61 @@ impl ModuleConfig {
 
         match config {
             None | Some(ModuleConfig::Es6) => {
-                let base_pass = import_hoister();
                 if paths.is_empty() {
-                    Box::new(base_pass)
+                    Box::new(noop())
                 } else {
                     let resolver = build_resolver(base_url, paths);
 
-                    Box::new(chain!(import_rewriter(base, resolver), base_pass))
+                    Box::new(import_rewriter(base, resolver))
                 }
             }
             Some(ModuleConfig::CommonJs(config)) => {
+                let base_pass = module_hoister();
                 if paths.is_empty() {
-                    Box::new(modules::common_js::common_js(
-                        root_mark,
-                        config,
-                        Some(scope),
+                    Box::new(chain!(
+                        base_pass,
+                        modules::common_js::common_js(root_mark, config, Some(scope),)
                     ))
                 } else {
                     let resolver = build_resolver(base_url, paths);
-
-                    Box::new(modules::common_js::common_js_with_resolver(
-                        resolver,
-                        base,
-                        root_mark,
-                        config,
-                        Some(scope),
+                    Box::new(chain!(
+                        base_pass,
+                        modules::common_js::common_js_with_resolver(
+                            resolver,
+                            base,
+                            root_mark,
+                            config,
+                            Some(scope),
+                        )
                     ))
                 }
             }
             Some(ModuleConfig::Umd(config)) => {
+                let base_pass = module_hoister();
```

Comment:
Seems like this let can be moved to the `else` block?

Reply:
Here is wrong.
But it should also be handled in the if clause, right?

Reply:
What do you mean by here is wrong?

Reply:
I just updated it.